### PR TITLE
#122 unit & UI test welcome page

### PR DESCRIPTION
**Pull request by ShawnGuerra-Bautista**
Issues: #122 

Tasks done:
1. Fixed the Handler() deprecated issues on WelcomePageActivity
2. Added UI test on Welcome page to check if the WelcomePageActivity will lead the user to the LoginPageActivity
3. Added Unit test for both MainActivity and WelcomePageActivity; Checks MainActivity goes to WelcomePageActivity and checks WelcomePageActivity goes to LoginPageActivity

These aren't complex tests. Nevertheless, they characterize the behaviour of components, so it's useful tests to know the actual behaviour of our components.

### CODE REVIEW

File: app/src/androidTest/java/com/example/bigowlapp/activity/WelcomePageActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+public class WelcomePageActivityTest {
+
+    @Rule
+    public ActivityScenarioRule<WelcomePageActivity> mActivityScenario
+            = new ActivityScenarioRule<>(WelcomePageActivity.class);
+
+    @Test
+    public void executeWelcomePageAtStartTest() {
+        Espresso.onView(isRoot()).perform(waitFor(3000));
+        Espresso.onView(withId(R.id.button)).check(matches(isDisplayed()));
```
Comment (refatK) [inline app/src/androidTest/java/com/example/bigowlapp/activity/WelcomePageActivityTest.java:35]: Maybe would be better to explicitly check it switched to the correct activity, but this works too.

File: app/src/test/java/com/example/bigowlapp/activity/MainActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.robolectric.annotation.Config;
+
+import static org.junit.Assert.assertTrue;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = Build.VERSION_CODES.N_MR1)
+public class MainActivityTest {
+    @Test
+    public void MainActivityGoesToWelcomePageTest() {
```
Comment (refatK) [inline app/src/test/java/com/example/bigowlapp/activity/MainActivityTest.java:20]: naming convention, should be lowercase

File: app/src/test/java/com/example/bigowlapp/activity/WelcomePageActivityTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.robolectric.shadows.ShadowLooper;
+
+import static org.junit.Assert.assertTrue;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = Build.VERSION_CODES.N_MR1)
+public class WelcomePageActivityTest {
+    @Test
+    public void WelcomePageGoesToLoginPageActivityTest() {
```
Comment (refatK) [inline app/src/test/java/com/example/bigowlapp/activity/WelcomePageActivityTest.java:21]: naming convention, should be lowercase

**Review by refatK — CHANGES_REQUESTED** [review]
Really good, and clean tests that check the intended behavior of WelcomePageActivity. Just some minor style complaints.

**Review by refatK — APPROVED** [review]
Thanks for changes

**Review by KennanHoa — APPROVED** [review]
All test passed

**Pull request merged**